plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -331,10 +331,12 @@ protected ResultSet getColumns(JdbcTableHandle tableHandle, DatabaseMetaData met
             throws SQLException
     {
         RemoteTableName remoteTableName = tableHandle.getRequiredNamedRelation().getRemoteTableName();
+        String escapePattern = metadata.getSearchStringEscape();
+        String tableName = remoteTableName.getTableName();
         return metadata.getColumns(
                 remoteTableName.getCatalogName().orElse(null),
-                escapeNamePattern(remoteTableName.getSchemaName(), metadata.getSearchStringEscape()).orElse(null),
-                escapeNamePattern(Optional.of(remoteTableName.getTableName()), metadata.getSearchStringEscape()).orElse(null),
+                remoteTableName.getSchemaName().map(name -> escapeNamePattern(name, escapePattern)).orElse(null),
+                (tableName == null) ? null : escapeNamePattern(tableName, escapePattern),
```

Comment:
Table name was not nullable previously as `Optional.of` would fail

Reply:
fixed, IMO explicit check would do